### PR TITLE
Add auxiliary tests

### PR DESCRIPTION
**Pull request by ltseeley**
Add message parsing, log, state, and initialization tests

### CODE REVIEW

File: src/test_helpers.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2018 Bitwise IO, Inc.
```
Comment (agunde406) [inline src/test_helpers.rs]: Should this be Cargill and 2019?

File: src/message_type.rs [hunk 2]
```diff
@@ -268,3 +268,161 @@ impl From<PbftMessageType> for String {
         format!("{:?}", msg_type)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    /// PBFT-04: A valid ParsedMessage can be constructed from a PbftMessage, a a PbftSignedVote,
```
Comment (agunde406) [inline src/message_type.rs]: double "a"

File: src/message_log.rs [hunk 3]
```diff
@@ -207,103 +207,111 @@ impl PbftLog {
...
-        ParsedMessage::from_pbft_message(msg).expect("Failed to parse PbftMessage")
-    }
-
-    /// Obtain the PeerId for node `which`
-    fn get_peer_id(cfg: &PbftConfig, which: usize) -> PeerId {
-        cfg.peers[which].clone()
-    }
-
-    /// Test that adding one message works as expected
+    /// PBFT-02: The log properly stores and retrieves blocks
```
[the remark this answers is not in the document]
Comment (ltseeley) [reply src/message_log.rs]: @vaporos had suggested that we include the identifiers so it's easy to reference the motivation/intent of the test. Shawn, what are your thoughts?

[the remark this answers is not in the document]
Comment (ltseeley) [reply src/message_log.rs]: After a conversation with Shawn, we agreed to remove external references from the unit tests in favor of including more detailed information with the tests.

**Pull request merged**